matchzoo/models/match_pyramid.py

```diff
@@ -0,0 +1,168 @@
+"""An implementation of MatchPyramid Model."""
+import typing
+
+import torch
+import torch.nn as nn
+
+from matchzoo.engine.param_table import ParamTable
+from matchzoo.engine.param import Param
+from matchzoo.engine.base_model import BaseModel
+from matchzoo.engine import hyper_spaces
+from matchzoo.modules import Matching
+from matchzoo.dataloader import callbacks
+from matchzoo.utils import parse_activation
+
+
+class MatchPyramid(BaseModel):
+    """
+    MatchPyramid Model.
+
+    Examples:
+        >>> model = MatchPyramid()
+        >>> model.params['embedding_output_dim'] = 300
+        >>> model.params['kernel_count'] = [16, 32]
+        >>> model.params['kernel_size'] = [[3, 3], [3, 3]]
+        >>> model.params['dpool_size'] = [3, 10]
+        >>> model.guess_and_fill_missing_params(verbose=0)
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params(with_embedding=True)
+        params.add(Param(name='kernel_count', value=[32],
+                         desc="The kernel count of the 2D convolution "
+                              "of each block."))
+        params.add(Param(name='kernel_size', value=[[3, 3]],
+                         desc="The kernel size of the 2D convolution "
+                              "of each block."))
+        params.add(Param(name='activation', value='relu',
+                         desc="The activation function."))
+        params.add(Param(name='dpool_size', value=[3, 10],
+                         desc="The max-pooling size of each block."))
+
+        params.add(Param(
+            'dropout_rate', 0.0,
+            hyper_space=hyper_spaces.quniform(
+                low=0.0, high=0.8, q=0.01),
+            desc="The dropout rate."
+        ))
+        return params
+
+    @classmethod
+    def get_default_padding_callback(
```

Comment:
MatchPyramid model does not need fixed padding length. Default get_default_padding_callback function is ok.